src/internal/pachsql/db.go

```diff
@@ -58,6 +92,85 @@ func OpenURL(u URL, password string) (*DB, error) {
 	return res, errors.EnsureStack(err)
 }
 
+// ListTables returns an array of SchemaTable structs that represent the tables.
+func ListTables(ctx context.Context, db *DB) ([]SchemaTable, error) {
+	rows, err := db.QueryContext(ctx, listTablesQuery)
+	if err != nil {
+		return nil, errors.Wrap(err, "could not list tables")
+	}
+	defer rows.Close()
+	var tables []SchemaTable
+	if err := sqlx.StructScan(rows, &tables); err != nil {
+		return nil, errors.Wrap(err, "could not scan tables")
+	}
+	return tables, nil
+}
+
+// ScannedRows queries db and calls MapScanRows on the response rows.
+func ScannedRows(ctx context.Context, db *DB, query string) ([]RowMap, error) {
+	rows, err := db.QueryContext(ctx, query)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to query database")
+	}
+	defer rows.Close()
+	scannedRows, err := MapScanRows(rows)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to scan rows")
+	}
+	return scannedRows, nil
+}
+
+// MapScanRows iterates through rows and returns an array of sqlx.MapScan results. This is useful
+// for when you want to capture the results, but don't know what columns the query can return
+// or when you don't want to create a struct to represent the columns.
+func MapScanRows(rows *sql.Rows) ([]RowMap, error) {
+	var processedRows []RowMap
+	for rows.Next() {
+		row := RowMap{}
+		if err := sqlx.MapScan(rows, row); err != nil {
+			return nil, errors.EnsureStack(err)
+		}
+		processedRows = append(processedRows, row)
+	}
+	return processedRows, nil
+}
+
+// KeyColumn returns the 'key' column if it exists for SchemaTable table. Otherwise, it queries the db and returns
+// the column_name where ordinal_position is 1. This column can be used to order future queries against this table.
```

Comment:
Thinking about this a little more, if you guess this wrong, then there won't be an index that lets the database engine retrieve the rows in this order, so it will have to sort them all before returning them to you.  To be honest, I'm not sure why the debug output needs to be in a particular order.

Reply:
Yeah that's true, we probably don't need to sort the response at all and can just iterate via rows.Next()